### PR TITLE
Update Gradle to 3.5 and Checkstyle to 7.7.

### PR DESCRIPTION
**Pull request by artem-zinnatullin**

### CODE REVIEW

File: gradle/checkstyle.gradle [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    configFile rootProject.file('checkstyle/checkstyle.xml')
+
+    source 'src'
+    include '**/*.java'
+
+    classpath = files()
+}
+
+checkstyle {
+    toolVersion = '7.7'
```
Comment (artem-zinnatullin) [inline gradle/checkstyle.gradle]: Had to update Checkstyle and remove one rule https://discuss.gradle.org/t/checkstyle-checker-cannot-be-created-with-gradle-3-5/22474/2

File: gradle/checkstyle.gradle [hunk 2]
```diff
@@ -0,0 +1,17 @@
+apply plugin: 'checkstyle'
```
Comment (artem-zinnatullin) [inline gradle/checkstyle.gradle]: Decided to extract it to common script to remove duplication

File: .gitignore [hunk 3]
```diff
@@ -18,6 +18,8 @@ target/
 
 # Local configuration file (sdk path, etc)
 local.properties
+.mainframer/
```
Comment (artem-zinnatullin) [inline .gitignore]: Sorry, but I just can't wait 10 minutes to build it on my mbp

File: gradle/checkstyle.gradle (hunk 1 above)
Comment (artem-zinnatullin) [inline gradle/checkstyle.gradle]: Ah and fix warnings detected by newer Checkstyle

File: .gitignore (hunk 3 above)
Comment (nikitin-da) [inline .gitignore]: 👍

**Comment by nikitin-da** [comment]
Changelogs: [Checkstyle](http://checkstyle.sourceforge.net/releasenotes.html), [Gradle](https://docs.gradle.org/current/release-notes.html)

File: storio-common/src/main/java/com/pushtorefresh/storio/TypeMappingFinder.java [hunk 4]
```diff
@@ -12,7 +12,7 @@
  */
 public interface TypeMappingFinder {
     @Nullable
-    <T> TypeMapping<T> findTypeMapping(@NonNull final Class<T> type);
+    <T> TypeMapping<T> findTypeMapping(@NonNull Class<T> type);
```
Comment (nikitin-da) [inline storio-common/src/main/java/com/pushtorefresh/storio/TypeMappingFinder.java]: :neutral_face:

**Pull request merged**